common/metrics/defs.go

```diff
@@ -3416,6 +3437,13 @@ var MetricDefs = map[ServiceIdx]map[int]metricDefinition{
 		DiagnosticsWorkflowSuccess:                    {metricName: "diagnostics_workflow_success", metricType: Counter},
 		DiagnosticsWorkflowExecutionLatency:           {metricName: "diagnostics_workflow_execution_latency", metricType: Timer},
 	},
+	ShardDistributor: {
+		ShardDistributorRequests:                 {metricName: "shard_distributor_requests", metricType: Counter},
+		ShardDistributorErrContextTimeoutCounter: {metricName: "shard_distributor_err_context_timeout", metricType: Counter},
+		ShardDistributorFailures:                 {metricName: "shard_distributor_failures", metricType: Counter},
+		ShardDistributorLatency:                  {metricName: "shard_distributor_latency", metricType: Timer},
+		ShardDistributorErrNamespaceNotFound:     {metricName: "shard_distributor_err_namespace_not_found", metricType: Counter},
```

Comment:
Why this err_ is special? I wonder it is not a tag for general "error".

Reply:
In Cadence we also have separate metrics for each known user error. I have used this some times to investigate problems so I think it's a good pattern.